tests/atest/rules/naming/hyphen-in-variable-name/test.robot

```diff
@@ -0,0 +1,19 @@
+*** Test Cases ***
+My Test Case
+    ${r}    Set Variable  ${2-1}  # this is fine
+    ${a-b}  Set Variable  1     # this will warn - because if it's later used as ${a-b} it can lead to ambiguous results
+    ${a\-b}  Set Variable  1    # this will warn - because if it's later used as ${a-b} it can lead to ambiguous results
+    ${-}    Set Variable  1     # this will warn - because if it's later used as ${a-b} it can lead to ambiguous results
+    ${a-}   Set Variable  1     # this will warn - because if it's later used as ${a-b} it can lead to ambiguous results
+    ${-b}   Set Variable  1     # this will warn - because if it's later used as ${a-b} it can lead to ambiguous results
+    My Keyword    1  2  3
+
+
+*** Keywords ***
+My Keyword
+    ${r}    Set Variable  ${2-1}  # this is fine
+    ${a-b}  Set Variable  1     # this will warn - because if it's later used as ${a-b} it can lead to ambiguous results
+    ${a\-b}  Set Variable  1    # this will warn - because if it's later used as ${a-b} it can lead to ambiguous results
```

Comment:
The point was that we probably should not var on hypen if its properly escaped. What do you think? We could go with it now but then update in next release

Reply:
I would keep it as it is because this is how RF understands it.